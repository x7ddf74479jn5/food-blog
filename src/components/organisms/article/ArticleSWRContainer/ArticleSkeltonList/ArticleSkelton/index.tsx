import { Skelton } from "@/components/atoms/Skelton";

export const ArticleSkelton: React.FC = () => {
  return (
    <div>
      <Skelton className="mb-4 aspect-video w-full" />
      <Skelton className="mb-4 h-8 w-4/5" />
      <div className="flex flex-col gap-2">
        <Skelton className="h-4 w-full" />
        <Skelton className="h-4 w-32" />
      </div>
    </div>
  );
};
