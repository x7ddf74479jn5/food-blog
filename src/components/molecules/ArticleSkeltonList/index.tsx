import { ArticleSkelton } from "@/components/atoms/ArticleSkelton";

export const ArticleSkeltonList: React.VFC = () => {
  return (
    <div className="space-y-12">
      <ArticleSkelton />
      <ArticleSkelton />
      <ArticleSkelton />
      <ArticleSkelton />
      <ArticleSkelton />
    </div>
  );
};
