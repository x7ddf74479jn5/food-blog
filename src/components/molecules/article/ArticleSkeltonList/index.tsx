import { ArticleSkelton } from "../ArticleSkelton";

export const ArticleSkeltonList: React.FC = () => {
  const listItems = [...Array(5)].map((_, index) => (
    <li key={index}>
      <ArticleSkelton />
    </li>
  ));

  return <ul className="space-y-12">{listItems}</ul>;
};
