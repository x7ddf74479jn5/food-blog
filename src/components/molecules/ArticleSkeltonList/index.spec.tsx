import renderer from "react-test-renderer";

import { ArticleSkeltonList } from "../ArticleSkeltonList";

describe("components/molecules/ArticleSkeltonList", () => {
  it("snapshot", () => {
    const tree = renderer.create(<ArticleSkeltonList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
