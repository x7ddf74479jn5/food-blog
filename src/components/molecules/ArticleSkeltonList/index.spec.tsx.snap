// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/ArticleSkeltonList snapshot 1`] = `
<div
  className="space-y-12"
>
  <article
    className="w-full shadow-lg animate-pulse select-none"
  >
    <div
      className="mb-4 w-full bg-gray-300 aspect-w-16 aspect-h-9"
    />
    <div
      className="mb-4 w-4/5 h-8 bg-gray-300 "
    />
    <div
      className="flex flex-col gap-2"
    >
      <div
        className="w-full h-4 bg-gray-300 "
      />
      <div
        className="w-32 h-4 bg-gray-300 "
      />
    </div>
  </article>
  <article
    className="w-full shadow-lg animate-pulse select-none"
  >
    <div
      className="mb-4 w-full bg-gray-300 aspect-w-16 aspect-h-9"
    />
    <div
      className="mb-4 w-4/5 h-8 bg-gray-300 "
    />
    <div
      className="flex flex-col gap-2"
    >
      <div
        className="w-full h-4 bg-gray-300 "
      />
      <div
        className="w-32 h-4 bg-gray-300 "
      />
    </div>
  </article>
  <article
    className="w-full shadow-lg animate-pulse select-none"
  >
    <div
      className="mb-4 w-full bg-gray-300 aspect-w-16 aspect-h-9"
    />
    <div
      className="mb-4 w-4/5 h-8 bg-gray-300 "
    />
    <div
      className="flex flex-col gap-2"
    >
      <div
        className="w-full h-4 bg-gray-300 "
      />
      <div
        className="w-32 h-4 bg-gray-300 "
      />
    </div>
  </article>
  <article
    className="w-full shadow-lg animate-pulse select-none"
  >
    <div
      className="mb-4 w-full bg-gray-300 aspect-w-16 aspect-h-9"
    />
    <div
      className="mb-4 w-4/5 h-8 bg-gray-300 "
    />
    <div
      className="flex flex-col gap-2"
    >
      <div
        className="w-full h-4 bg-gray-300 "
      />
      <div
        className="w-32 h-4 bg-gray-300 "
      />
    </div>
  </article>
  <article
    className="w-full shadow-lg animate-pulse select-none"
  >
    <div
      className="mb-4 w-full bg-gray-300 aspect-w-16 aspect-h-9"
    />
    <div
      className="mb-4 w-4/5 h-8 bg-gray-300 "
    />
    <div
      className="flex flex-col gap-2"
    >
      <div
        className="w-full h-4 bg-gray-300 "
      />
      <div
        className="w-32 h-4 bg-gray-300 "
      />
    </div>
  </article>
</div>
`;
